src/realm/util/encrypted_file_mapping.cpp

```diff
@@ -371,6 +371,9 @@ void EncryptedFileMapping::mark_up_to_date(size_t i) noexcept
 
 bool EncryptedFileMapping::copy_up_to_date_page(size_t page) noexcept
 {
+    // Precondition: this method must never be called for a page which
+    // is already up to date.
+    REALM_ASSERT(!m_up_to_date_pages[page]);
```

Comment:
Just out of curiosity, how hot is this function? Should we consider strengthening the assert to be `REALM_ASSERT_RELEASE`? As a reminder, this non-qualified assert is enabled only for bindings that chose to do so.